libs/langgraph/tests/test_prebuilt.py

```diff
@@ -368,6 +374,39 @@ def tool2(some_val: int) -> str:
         create_react_agent(model.bind_tools([tool1]), [tool2])
 
 
+def test__infer_handled_types() -> None:
+    def handle(e):  # type: ignore
+        return ""
+
+    def handle2(e: Exception) -> str:
+        return ""
+
+    def handle3(e: Union[ValueError, ToolException]) -> str:
```

Comment:
What happens if the function is misstyped?

```Union[Union[ValueError, TypeError], ToolException]]```

or
```Tuple[ValueError, TypeError]```

Reply:
will break -- good catch

Reply:
added a better error message for those situations